pkg/service/oidc4vc/oidc4vc_service_exchnage_code_test.go

```diff
@@ -0,0 +1,152 @@
+	}()
+	baseTx := &oidc4vc.Transaction{
+		ID: oidc4vc.TxID("id"),
+		TransactionData: oidc4vc.TransactionData{
+			TokenEndpoint:  fmt.Sprintf("http://%s/token", addr),
+			IssuerAuthCode: authCode,
+		},
+	}
+	store.EXPECT().FindByOpState(gomock.Any(), opState).Return(baseTx, nil)
+	store.EXPECT().Update(gomock.Any(), gomock.Any()).
+		DoAndReturn(func(ctx context.Context, tx *oidc4vc.Transaction) error {
+			assert.Equal(t, baseTx, tx)
+			assert.Equal(t, "SlAV32hkKG", tx.IssuerToken)
+
+			return nil
+		})
+
+	resp, err := srv.ExchangeAuthorizationCode(context.TODO(), opState)
+	assert.NoError(t, err)
+	assert.NotEmpty(t, resp)
+}
+
+func TestExchangeCodeErrFindTx(t *testing.T) {
+	store := NewMockTransactionStore(gomock.NewController(t))
+	srv, err := oidc4vc.NewService(&oidc4vc.Config{TransactionStore: store})
+	assert.NoError(t, err)
+
+	store.EXPECT().FindByOpState(gomock.Any(), gomock.Any()).Return(nil, errors.New("tx not found"))
+	resp, err := srv.ExchangeAuthorizationCode(context.TODO(), "123")
+	assert.Empty(t, resp)
+	assert.ErrorContains(t, err, "tx not found")
+}
+
+func TestExchangeCodeIssuerError(t *testing.T) {
+	store := NewMockTransactionStore(gomock.NewController(t))
+	srv, err := oidc4vc.NewService(&oidc4vc.Config{TransactionStore: store})
+	assert.NoError(t, err)
+
+	addr := "127.0.0.1:65122"
+	httpSrv := &http.Server{Addr: addr}
+
+	go func() {
+		mux := http.NewServeMux()
+		mux.HandleFunc("/token", func(writer http.ResponseWriter, request *http.Request) {
+			_, _ = writer.Write([]byte("go away"))
+			writer.WriteHeader(500)
+		})
+		httpSrv.Handler = mux
+		_ = httpSrv.ListenAndServe()
+	}()
+
+	time.Sleep(2 * time.Second)
+	defer func() {
+		_ = httpSrv.Close()
+	}()
+
+	store.EXPECT().FindByOpState(gomock.Any(), gomock.Any()).Return(&oidc4vc.Transaction{
+		TransactionData: oidc4vc.TransactionData{
+			TokenEndpoint: fmt.Sprintf("http://%s/token", addr),
+		},
+	}, nil)
+
+	resp, err := srv.ExchangeAuthorizationCode(context.TODO(), "sadsadas")
+	assert.Empty(t, resp)
+	assert.ErrorContains(t, err, "oauth2: server response missing access_token")
+}
+
+func TestExchangeCodeStoreUpdateErr(t *testing.T) {
+	store := NewMockTransactionStore(gomock.NewController(t))
+	srv, err := oidc4vc.NewService(&oidc4vc.Config{TransactionStore: store})
+	assert.NoError(t, err)
+
+	opState := uuid.NewString()
+	authCode := uuid.NewString()
+
+	addr := "127.0.0.1:65122"
+	httpSrv := &http.Server{Addr: addr}
+
+	go func() {
+		mux := http.NewServeMux()
+		mux.HandleFunc("/token", func(writer http.ResponseWriter, request *http.Request) {
+			writer.Header().Set("content-type", "application/json")
+			writer.WriteHeader(200)
+			_, _ = writer.Write([]byte("{   \"access_token\": \"SlAV32hkKG\",   \"token_type\": \"Bearer\",   \"refresh_token\": \"8xLOxBtZp8\",   \"expires_in\": 3600,   \"id_token\": \"1234\"  }")) // nolint
+		})
+		httpSrv.Handler = mux
+		_ = httpSrv.ListenAndServe()
+	}()
+
+	time.Sleep(2 * time.Second)
```

Comment:
this doesn't look like a unit test
try to introduce abstraction for `oauthConfig` with its `Exchange` method and use mock